src/components/HelFormFields/HelDateTimeField.js

```diff
@@ -8,55 +8,52 @@ import HelTimePicker from './HelTimePicker.js'
 import {connect} from 'react-redux'
 import {setData, updateSubEvent} from 'src/actions/editor.js'
 
-import { FormattedMessage } from 'react-intl'
+import {FormattedMessage} from 'react-intl'
 
 import validationRules from 'src/validation/validationRules.js';
 import ValidationPopover from 'src/components/ValidationPopover'
 
 import moment from 'moment'
 
-const HelDateTimeField = React.createClass({
-
-    getInitialState: function() {
+class HelDateTimeField extends React.Component {
+    constructor(props) {
+        super(props)
+        
         let defaultValue = this.props.defaultValue || null
         if(moment(defaultValue).isValid()) {
             defaultValue = moment(defaultValue).tz('Europe/Helsinki');
-            return {
+            this.state = {
                 date: defaultValue,
-                time: defaultValue.format('HH.mm')
+                time: defaultValue.format('HH.mm'),
+            }
+        } else {
+            this.state = {
+                date: null,
+                time: null,
             }
         }
 
-        return {
-            date: null,
-            time: null
-        }
-    },
-
-    propTypes: {
-        name: PropTypes.string.isRequired,
-        eventKey: PropTypes.string
-    },
-
-    contextTypes: {
-        intl: PropTypes.object,
-        dispatch: PropTypes.func
-    },
+        this.onChange = this.onChange.bind(this)
+        this.onBlur = this.onBlur.bind(this)
+        this.getDateTimeFromFields = this.getDateTimeFromFields.bind(this)
+        this.parseValueFromString = this.parseValueFromString.bind(this)
+        this.getValidationErrors = this.getValidationErrors.bind(this)
```

Comment:
I don't think `getDateTimeFromFields`, `parseValueFromString` and `getValidationErrors` need to be binded since they never refer to `this`.